### PR TITLE
Added code for processing GSheets

### PR DESCRIPTION
**Pull request by KarloVis**
1. Added rules in Heartbeat.drl to connect to the spreadsheet
2. Made a function in Smith.java to process the data (for now just shows you the data)

### CODE REVIEW

File: services/rule_engine/src/main/java/agents/Smith.java [hunk 1]
```diff
@@ -1,14 +1,27 @@
 package agents;
```
[the remark this answers is not in the document]
Comment (KarloVis) [reply services/rule_engine/src/main/java/agents/Smith.java:1]: I didn't change anything in those files, so I don't know what exactly is missing.

File: services/rule_engine/src/main/java/agents/Smith.java [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
 import com.mindsmiths.ruleEngine.model.Agent;
+import com.mindsmiths.ruleEngine.util.Log;
+
+import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.ToString;
 
 @Data
 @ToString(callSuper = true)
+@AllArgsConstructor
```
[the remark this answers is not in the document]
Comment (KarloVis) [reply services/rule_engine/src/main/java/agents/Smith.java:15]: When I add that constructor, it complains about the Smith() method in Smith() class and I don't know how to fix that

File: services/rule_engine/src/main/resources/rules/smith/Heartbeat.drl [hunk 3]
```diff
@@ -1,13 +1,29 @@
...
 
+import com.mindsmiths.gsheetsAdapter.GSheetsAdapterAPI
+import com.mindsmiths.gsheetsAdapter.reply.Spreadsheet
+
+rule "Get Google Sheets"
+    when
+        Heartbeat(now: timestamp) from entry-point "signals"
+        agent: Smith(/*lastTableUpdate == null || lastTableUpdate before[30s] now*/)
+    then
+        agent.lastTableUpdate = now;
```
[the remark this answers is not in the document]
Comment (KarloVis) [reply services/rule_engine/src/main/resources/rules/smith/Heartbeat.drl:18]: That didn't work the first time I was coding it, but now it works so I've fixed it.

File: services/rule_engine/src/main/java/agents/Smith.java [hunk 4]
```diff
@@ -1,14 +1,27 @@
...
 public class Smith extends Agent {
 
+    public LocalDateTime lastTableUpdate = null;
```
[the remark this answers is not in the document]
Comment (KarloVis) [reply services/rule_engine/src/main/java/agents/Smith.java:18]: Done

File: services/rule_engine/src/main/java/agents/Smith.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjuraj) [reply services/rule_engine/src/main/java/agents/Smith.java:1]: Can you run `gsheets-adapter setup` again and see if it makes the changes?

**Pull request merged**